lib/jnpr/junos/utils/sw.py

```diff
@@ -1099,6 +1088,57 @@ def reboot(self, in_min=0, at=None, all_re=True, on_node=None, vmhost=False):
                     if output_msg is not "":
                         got = output_msg
             return got
+        except Exception as err:
+            raise err
+
+    # -------------------------------------------------------------------------
+    # reboot - system reboot
+    # -------------------------------------------------------------------------
+    def reboot(
+        self, in_min=0, at=None, all_re=True, on_node=None, vmhost=False, other_re=False
+    ):
+        """
+        Perform a system reboot, with optional delay (in minutes) or at
+        a specified date and time.
+
+        If the device is equipped with dual-RE, then both RE will be
+        rebooted.  This code also handles EX/QFX VC.
+
+        :param int in_min: time (minutes) before rebooting the device.
+
+        :param str at: date and time the reboot should take place. The
+            string must match the junos cli reboot syntax
+
+        :param bool all_re: In case of dual re or VC setup, function by default
+            will reboot all. If all is False will only reboot connected device
+
+        :param str on_node: In case of linux based device, function will by default
+            reboot the whole device. If any specific node is mentioned,
+            reboot will be performed on mentioned node
+
+        :param bool vmhost:
+            (Optional) A boolean indicating to run 'request vmhost reboot'.
+            The default is ``vmhost=False``.
+
+        :param str other_re: If the system has dual Routing Engines and this option is C(true),
+            then the action is performed on the other REs in the system.
+
+        :returns:
+            * reboot message (string) if command successful
+        """
+        if self._dev.facts["_is_linux"]:
+            if on_node is None:
+                cmd = E("request-shutdown-reboot")
+            else:
+                cmd = E("request-node-reboot")
+                cmd.append(E("node", on_node))
+        elif vmhost is True:
+            cmd = E("request-vmhost-reboot")
+        else:
+            cmd = E("request-reboot")
+
+        try:
+            return self._system_operation(cmd, in_min, at, all_re, other_re, vmhost)
```

Comment:
Kindly follow the same parameter  order which you followed in reboot call line no 1098

self, in_min=0, at=None, all_re=True, on_node=None, vmhost=False, other_re=False

Reply:
All the new parameter are added at the end, otherwise people having current scripts will get wrong output when they execute.